portal/components/react-admin/base/resources/teachers.js

```diff
@@ -195,8 +195,8 @@ export const TeacherList = (props) => {
     >
       {isSmall ? (
         <SimpleList
-          primaryText={(record) => record.name}
-          secondaryText={(record) => record.first_name}
+          primaryText={(record) => record.user.username}
+          secondaryText={(record) => record.user.first_name}
```

Comment:
@Shruti-Agarwl Can you ensure this reference is null-safe, since it's possible that `record.user` will be undefined in rare cases?

Reply:
Added the conditional check @bharatkashyap